internal/auth/authn/jwtverifer.go

```diff
@@ -0,0 +1,47 @@
+package authn
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+type JWTAuth struct {
+	JwksUrl          string
+	OidcDiscoveryUrl string
+}
+
+func (j JWTAuth) ValidateToken(ctx context.Context, token string) (bool, error) {
+	jwkSet, err := jwk.Fetch(ctx, j.JwksUrl)
+	if err != nil {
+		return false, err
+	}
+	_, err = jwt.Parse([]byte(token), jwt.WithKeySet(jwkSet), jwt.WithValidate(true))
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (j JWTAuth) GetTokenRequestURL(ctx context.Context) (string, error) {
+	res, err := http.Get(j.OidcDiscoveryUrl)
+	if err != nil {
+		return "", err
```

Comment:
would it make sense to add more context on these errors to make it easier to debug?

Reply:
updated